Address safer CPP warnings in PageClientImplMac

csaavedra opened the pull request:
#### 008b65cc75bf2572e6167acddee1de091db127fb
<pre>
Address safer CPP warnings in PageClientImplMac
<a href="https://bugs.webkit.org/show_bug.cgi?id=300023">https://bugs.webkit.org/show_bug.cgi?id=300023</a>

Reviewed by Chris Dumez.

Not unskipping the file due to a few false-positives
that should be addressed by recent changes in the checker
and a few that are addressed in a separate MR.

* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::viewSize):
(WebKit::PageClientImpl::activeWindow const):
(WebKit::PageClientImpl::assistiveTechnologyMakeFirstResponder):
(WebKit::PageClientImpl::makeFirstResponder):
(WebKit::PageClientImpl::isMainViewVisible):
(WebKit::PageClientImpl::isViewVisibleOrOccluded):
(WebKit::PageClientImpl::registerInsertionUndoGrouping):
(WebKit::PageClientImpl::canUndoRedo):
(WebKit::PageClientImpl::executeUndoRedo):
(WebKit::PageClientImpl::convertToDeviceSpace):
(WebKit::PageClientImpl::convertToUserSpace):
(WebKit::PageClientImpl::screenToRootView):
(WebKit::PageClientImpl::rootViewToScreen):
(WebKit::PageClientImpl::rootViewToWindow):
(WebKit::PageClientImpl::doneWithKeyEvent):
(WebKit::PageClientImpl::setRemoteLayerTreeRootNode):
(WebKit::PageClientImpl::wheelEventWasNotHandledByWebCore):
(WebKit::PageClientImpl::boundsOfLayerInLayerBackedWindowCoordinates const):
(WebKit::PageClientImpl::windowIsFrontWindowUnderMouse):
(WebKit::PageClientImpl::computeAutomaticTopObscuredInset):
(WebKit::PageClientImpl::makeViewBlank):

Canonical link: <a href="https://commits.webkit.org/301138@main">https://commits.webkit.org/301138@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/45333759cbe7f7e307569929eb502d1c1c55b771

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125054 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44722 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35460 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131902 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76931 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45413 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53287 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95190 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63161 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128008 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36247 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111840 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75733 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35162 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29993 "Build is in progress. Recent messages:") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75384 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106010 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30222 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134581 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51873 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39669 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103660 "2 flakes 56 failures") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108051 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103434 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26338 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48780 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27069 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48922 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51759 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57552 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51132 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54489 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52823 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
```diff
@@ -462,27 +462,30 @@ - (NSCursor *)_cursorRectCursor;
 {
-    NSPoint windowCoord = [[m_view window] convertPointFromScreen:point];
-    return IntPoint([m_view convertPoint:windowCoord fromView:nil]);
+    RetainPtr view { m_view.get() };
```
The pattern of using curly brackets like this for local variable initialization is a bit unusual. Initializing with `= ` is way more common and it'd be nice to follow our usual pattern in future patches for consistency, even though equivalent.

cdumez commented on Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
```diff
@@ -624,7 +627,7 @@ - (NSCursor *)_cursorRectCursor;
 void PageClientImpl::setRemoteLayerTreeRootNode(RemoteLayerTreeNode* rootNode)
 {
-    checkedImpl()->setAcceleratedCompositingRootLayer(rootNode ? rootNode->layer() : nil);
+    checkedImpl()->setAcceleratedCompositingRootLayer(rootNode ? RetainPtr { rootNode->layer() }.get() : nil);
```
Can call `protectedLayer().get()`

cdumez commented on Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
```diff
@@ -1033,15 +1036,15 @@ static inline NSCorrectionResponse toCorrectionResponse(AutocorrectionResponse r
 {
-    RetainPtr window = [m_view window];
-    if (([window styleMask] & NSWindowStyleMaskFullSizeContentView) && ![window titlebarAppearsTransparent] && ![m_view enclosingScrollView]) {
+    RetainPtr window = [m_view.get() window];
```
Please store `m_view.get()` in a local variable to avoid refcounting churn.

cdumez reviewed (approved):
LGTM with nits.

csaavedra committed:
Address safer CPP warnings in PageClientImplMac
https://bugs.webkit.org/show_bug.cgi?id=300023

Reviewed by Chris Dumez.

Not unskipping the file due to a few false-positives
that should be addressed by recent changes in the checker
and a few that are addressed in a separate MR.

* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::viewSize):
(WebKit::PageClientImpl::activeWindow const):
(WebKit::PageClientImpl::assistiveTechnologyMakeFirstResponder):
(WebKit::PageClientImpl::makeFirstResponder):
(WebKit::PageClientImpl::isMainViewVisible):
(WebKit::PageClientImpl::isViewVisibleOrOccluded):
(WebKit::PageClientImpl::registerInsertionUndoGrouping):
(WebKit::PageClientImpl::canUndoRedo):
(WebKit::PageClientImpl::executeUndoRedo):
(WebKit::PageClientImpl::convertToDeviceSpace):
(WebKit::PageClientImpl::convertToUserSpace):
(WebKit::PageClientImpl::screenToRootView):
(WebKit::PageClientImpl::rootViewToScreen):
(WebKit::PageClientImpl::rootViewToWindow):
(WebKit::PageClientImpl::doneWithKeyEvent):
(WebKit::PageClientImpl::setRemoteLayerTreeRootNode):
(WebKit::PageClientImpl::wheelEventWasNotHandledByWebCore):
(WebKit::PageClientImpl::boundsOfLayerInLayerBackedWindowCoordinates const):
(WebKit::PageClientImpl::windowIsFrontWindowUnderMouse):
(WebKit::PageClientImpl::computeAutomaticTopObscuredInset):
(WebKit::PageClientImpl::makeViewBlank):

Canonical link: https://commits.webkit.org/301138@main
